### ISSUE TITLE
Documentation lacks code samples and information about collections support

### ISSUE THREAD

**Issue by afayes** [kivy/pyjnius#295]
Documentation lacks code samples and information about collections support. Py4J documentation is far better.

I checked documentation at: http://pyjnius.readthedocs.io/en/latest/genindex.html

some of my findings below:

```
from jnius import autoclass

ArrayList = autoclass('java.util.ArrayList')
Integer = autoclass('java.lang.Integer')
integer = Integer(1)
ls = ArrayList()
ls.add("text") # works fine
# ls.add(5) # does not work - get exception
ls.add(integer) # fine
# ls[0]= "aaa" # does not work - get exception
print ls.get(0) # works fine
print ls[0] # works fine - didn't expect that - not documented

HashMap = autoclass('java.util.HashMap')
mp = HashMap()
mp.put("key", "value") # works fine
mp.put(integer, "value") # works fine
# mp.put(1, "value") does not work - throws exception
# mp.put("key", 5) # does not work - throws exception

print mp.get("key") # works fine
# print mp["key"] # does not work throws exception

```

**Comment by KeyWeeUsr** [comment]
This was a missing conversion on our side because although we handled coversion for `java.lang.Object`, we did that only for strings (no ints, floats, etc). It's fixed in #394. Also re the indexing and accessing the keys - PyJNIus is not a complete wrapper for Java, therefore just use the appropriate Java methods for that such as [`java.util.ArrayList.set()`](
https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#set-int-E-).
